From 8bf005358844f6cf35cbb12c70b623b4d33b6d87 Mon Sep 17 00:00:00 2001
From: Andy Fleming <afleming@freescale.com>
Date: Thu, 14 Jun 2012 14:50:12 -0500
Subject: [PATCH 24/57] t4: Set the PIR for threads

commit baf9c0579bf12644403244b3883c6ecac39ef7c0 from
https://git.freescale.com/git-private/cgit.cgi/ppc/t4240/linux.git

The PIR is used in a few places in the kernel to deal with messaging.
The threads come up in Linux, so we need to set the PIR in Linux for
the threads.

Signed-off-by: Andy Fleming <afleming@freescale.com>
---
 arch/powerpc/kernel/head_64.S |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/powerpc/kernel/head_64.S b/arch/powerpc/kernel/head_64.S
index 9c65e57..19d2d7a 100644
--- a/arch/powerpc/kernel/head_64.S
+++ b/arch/powerpc/kernel/head_64.S
@@ -171,6 +171,7 @@ _GLOBAL(fsl_secondary_thread_init)
 
 	mfspr	r3, SPRN_PIR
 	rlwimi	r3, r3, 30, 2, 30
+	mtspr	SPRN_PIR, r3
 #endif
 
 _GLOBAL(generic_secondary_thread_init)
-- 
1.7.9.7

